pkg/manager/controllers/managementaddonprogressing/controller.go

```diff
@@ -0,0 +1,238 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	addonv1alpha1 "open-cluster-management.io/api/addon/v1alpha1"
+	addonv1alpha1client "open-cluster-management.io/api/client/addon/clientset/versioned"
+	addoninformerv1alpha1 "open-cluster-management.io/api/client/addon/informers/externalversions/addon/v1alpha1"
+	addonlisterv1alpha1 "open-cluster-management.io/api/client/addon/listers/addon/v1alpha1"
+	clusterinformersv1beta1 "open-cluster-management.io/api/client/cluster/informers/externalversions/cluster/v1beta1"
+	clusterlisterv1beta1 "open-cluster-management.io/api/client/cluster/listers/cluster/v1beta1"
+	clusterv1beta1 "open-cluster-management.io/api/cluster/v1beta1"
+
+	"open-cluster-management.io/addon-framework/pkg/addonmanager/constants"
+	"open-cluster-management.io/addon-framework/pkg/basecontroller/factory"
+	"open-cluster-management.io/addon-framework/pkg/index"
+)
+
+// managementAddonProgressingController reconciles instances of clustermanagementaddon the hub
+// based to update related object and status condition.
+type managementAddonProgressingController struct {
+	addonClient                  addonv1alpha1client.Interface
+	managedClusterAddonLister    addonlisterv1alpha1.ManagedClusterAddOnLister
+	clusterManagementAddonLister addonlisterv1alpha1.ClusterManagementAddOnLister
+	placementLister              clusterlisterv1beta1.PlacementLister
+	placementDecisionLister      clusterlisterv1beta1.PlacementDecisionLister
+}
+
+func NewManagementAddonProgressingController(
+	addonClient addonv1alpha1client.Interface,
+	addonInformers addoninformerv1alpha1.ManagedClusterAddOnInformer,
+	clusterManagementAddonInformers addoninformerv1alpha1.ClusterManagementAddOnInformer,
+	placementInformer clusterinformersv1beta1.PlacementInformer,
+	placementDecisionInformer clusterinformersv1beta1.PlacementDecisionInformer,
+) factory.Controller {
+	c := &managementAddonProgressingController{
+		addonClient:                  addonClient,
+		managedClusterAddonLister:    addonInformers.Lister(),
+		clusterManagementAddonLister: clusterManagementAddonInformers.Lister(),
+	}
+
+	return factory.New().WithInformersQueueKeysFunc(
+		func(obj runtime.Object) []string {
+			accessor, _ := meta.Accessor(obj)
+			return []string{accessor.GetName()}
+		},
+		addonInformers.Informer(), clusterManagementAddonInformers.Informer()).
+		WithInformersQueueKeysFunc(index.ClusterManagementAddonByPlacementDecisionQueueKey(clusterManagementAddonInformers), placementDecisionInformer.Informer()).
+		WithInformersQueueKeysFunc(index.ClusterManagementAddonByPlacementQueueKey(clusterManagementAddonInformers), placementInformer.Informer()).
+		WithSync(c.sync).ToController("management-addon-status-controller")
+
+}
+
+func (c *managementAddonProgressingController) sync(ctx context.Context, syncCtx factory.SyncContext, addonName string) error {
```

Comment:
how do you think about putting this as a reconciler in addonConfiguration reconciler? It seems to me that since we have the graph built, getting to know the number of updating and set condition is easier?

Reply:
remove managementaddonprogressing from this PR. will create another PR to update cma progressing.